server/sonar-server/src/main/java/org/sonar/server/organization/OrganizationCreation.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.organization;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.api.web.UserRole;
+import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
+
+import static java.util.Objects.requireNonNull;
+
+public interface OrganizationCreation {
+  String OWNERS_GROUP_NAME = "Owners";
+  String OWNERS_GROUP_DESCRIPTION_PATTERN = "Owners of organization %s";
+  String PERM_TEMPLATE_DESCRIPTION_PATTERN = "Default permission template of organization %s";
+
+  /**
+   * Create a new Organization with the specified properties and of which the specified user will assign Administer
+   * Organization permission.
+   * <p>
+   * This method does several operations at once:
+   * <ol>
+   *   <li>create an ungarded organization with the specified details</li>
+   *   <li>create a group called {@link #OWNERS_GROUP_NAME Owners} with Administer Organization permission</li>
+   *   <li>make the specified user a member of this group</li>
+   *   <li>create a default template for the organization (which name and description will follow patterns
+   *       {@link #OWNERS_GROUP_NAME} and {@link #OWNERS_GROUP_DESCRIPTION_PATTERN} based on the organization name)</li>
+   *   <li>this group defines the specified permissions (which effectively makes projects public):
+   *     <ul>
+   *       <li>group {@link #OWNERS_GROUP_NAME Owners} : {@link UserRole#ADMIN ADMIN}</li>
+   *       <li>group {@link #OWNERS_GROUP_NAME Owners} : {@link UserRole#ISSUE_ADMIN ISSUE_ADMIN}</li>
+   *       <li>any one : {@link UserRole#USER USER}</li>
+   *       <li>any one : {@link UserRole#CODEVIEWER CODEVIEWER}</li>
+   *     </ul>
+   *   </li>
+   * </ol>
+   * </p>
+   *
+   * @return the created organization
+   *
+   * @throws KeyConflictException if an organization with the specified key already exists
+   * @throws IllegalArgumentException if any field of {@code newOrganization} is invalid according to {@link OrganizationValidation}
+   */
+  OrganizationDto create(DbSession dbSession, long createUserId, NewOrganization newOrganization) throws KeyConflictException;
+
+  final class KeyConflictException extends Exception {
```

Comment:
Why introducing a checked exception ? We are used to rely on unchecked exceptions only.